### PR TITLE
(untested) use the reason/demand format for getReject

### PR DESCRIPTION
**Pull request by wonderingabout**
this removes repetitionns in the conn_log and reject message
thus making reject messages more easily maintainable and standardized

### CODE REVIEW

File: connection.js [hunk 1]
```diff
@@ -685,19 +685,18 @@ function rankToString(r) {
     else          return `${30 - R}k`;     // R < 30:  1 kyu or weaker
 }
 
-function getRejectBanned(username, ranked) {
-    return getReject(`You (${username}) are not allowed to play ${ranked}${ranked ? " " : ""}games against this bot.`)
+function getReject(reason, demand) {
```
Comment (Dorus) [inline connection.js:688]: Give demand a default value:
```Js
 function getReject(reason, demand = "") {
    conn_log(reason);
    return { reject: true, msg: `${reason}${demand}`};
}
```

File: connection.js [hunk 2]
```diff
@@ -713,20 +712,16 @@ function get_r_u_arr_booleans(familyNameString, notificationRanked) {
 
 function getBooleans_r_u_Reject(argNameString, nameF, ending) {
     const rankedUnranked = beforeRankedUnrankedGamesSpecial("for ", "", argNameString, "");
-    const msg = `${nameF} not allowed on this bot ${rankedUnranked}${ending}.`;
-    conn_log(msg);
-    return { reject: true, msg };
+    return getReject(`${nameF} not allowed on this bot ${rankedUnranked}${ending}.`);
```
Comment (Dorus) [inline connection.js:715]: Do not forget to add unit testing on these changes. I dont think they are covered.

File: connection.js [hunk 3]
```diff
@@ -713,20 +712,16 @@ function get_r_u_arr_booleans(familyNameString, notificationRanked) {
...
-    const msg = `Your selected board size ${notificationWidth}x${notificationHeight} `
-                + `is not square, not allowed ${rankedUnranked}, `
-                + `please choose a SQUARE board size (same width and `
-                + `height), for example try 9x9 or 19x19}`;
-    return { reject: true, msg };
+    const reason = `Selected board size ${notificationWidth}x${notificationHeight} `
+                   + `is not square, not allowed ${rankedUnranked}`;
+    const demand = ` , please choose a SQUARE board size (same width and `
+                   + `height), for example try 9x9 or 19x19}`;
+    return getReject(reason, demand);
```
Comment (Dorus) [inline connection.js:724]: Especially here i want to see unit testing. For example, there seem to be a random extra space at the start of  `demand`.

File: connection.js [hunk 4]
```diff
@@ -763,13 +758,11 @@ function genericAllowedFamiliesReject(argNameString, notificationUnit) {
...
-    conn_log(`${argFamilySingularString} ${rankedUnranked} `
-             + `-${notificationUnitConverted}-, not in -${argValueString}- `);
-    const msg = `${argFamilySingularString} -${notificationUnitConverted}- `
-                + `is not allowed on this bot ${rankedUnranked}, please `
-                + `choose one of these allowed ${argFamilySingularString}s `
-                + `${rankedUnranked}: -${argValueString}-`;
-    return { reject: true, msg };
+    const reason = `${argFamilySingularString} -${notificationUnitConverted}- `
+                   + `is not allowed on this bot ${rankedUnranked}`;
+    const demand = ` , please choose one of these allowed ${argFamilySingularString}s `
```
Comment (Dorus) [inline connection.js:763]: Another redundant space at start of `demand`.

**Comment by Dorus** [comment]
Also wouldn't `suggestion` be a better name instead of `demand`?

File: connection.js (hunk 1 above)
Comment (wonderingabout) [reply connection.js]: great, i added it !

File: connection.js (hunk 2 above)
Comment (wonderingabout) [reply connection.js:714]: i'd like to do that, but would need time to get used to it, i definitely will try to do it sometime :100:

File: connection.js (hunk 3 above)
Comment (wonderingabout) [reply connection.js:723]: indeed

File: connection.js (hunk 4 above)
Comment (wonderingabout) [reply connection.js:762]: indeed, same as https://github.com/online-go/gtp2ogs/pull/286#discussion_r426189771

**Comment by wonderingabout** [comment]
> Also wouldn't `suggestion` be a better name instead of `demand`?

i liked how `demand` matched the length of `reason`
maybe `condition`?

**Comment by wonderingabout** [comment]
it looks like this is too messy, will close the pr and perhaps submit an enhanced version in the future

**Pull request closed without merging**